python/paddle/distributed/fleet/meta_parallel/pipeline_parallel.py

```diff
@@ -22,11 +22,47 @@
 from ..utils.hybrid_parallel_util import broadcast_sharding_parameters
 from ..utils.log_util import logger
 from ..meta_optimizers.dygraph_optimizer import HybridParallelOptimizer, HybridParallelGradScaler
-from .pp_utils import p2p_communication as p2p
+from paddle.fluid.framework import _in_legacy_dygraph, _non_static_mode, in_dygraph_mode
 import paddle.fluid.core as core
 
 __all__ = []
 
+_VIRTUAL_PP_WORLD_SIZE = None
+_VIRTUAL_PP_RANK = None
+
+_REAL_PP_WORLD_SIZE = None
+_REAL_PP_RANK = None
+
+
+def is_pipeline_first_stage(ignore_virtual=False):
```

Comment:
I think we may split 1F1B pipeline and interleave pipe into two class.

Reply:
Ok, nice point